### PR TITLE
Update to silex 2.0

### PR DESCRIPTION
**Pull request by hispavista**
Update the core to silex 2.0

### CODE REVIEW

File: src/Symlex/Router/TwigRouter.php [hunk 1]
```diff
@@ -99,11 +99,12 @@ public function route($routePrefix = '', $servicePrefix = 'controller.web.', $se
...
         };
 
-        $indexRequestHandler = function (Request $request) use ($app, $container, $servicePrefix, $servicePostfix, $webRequestHandler) {
-            return $webRequestHandler('index', $request, 'index');
+        $indexRequestHandler = function (Request $request, $action='index') use ($app, $container, $servicePrefix, $servicePostfix, $webRequestHandler) {
+            return $webRequestHandler('index', $request, $action);
         };
 
         $app->get($routePrefix . '/', $indexRequestHandler);
+        $app->get($routePrefix . '/{action}', $indexRequestHandler)->assert('action', '.+');
```
Comment (lastzero) [inline src/Symlex/Router/TwigRouter.php:107]: Wouldn't this prevent the rule in the next line from working? Looks like the same pattern or what would be the difference?

```
$routePrefix . '/{action}'
$routePrefix . '/{controller}'
```

Comment (hispavista) [reply src/Symlex/Router/TwigRouter.php:107]: Is to have more than one action in the IndexController. if IndexController have index and edit actions the routes "/" and "/edit" will be catched by IndexController and the other routes with index actions by their respectives controllers

Comment (lastzero) [reply src/Symlex/Router/TwigRouter.php:107]: The idea of this router is, that you create a new controller for each "page" and the request is routed to the indexAction, if there is no action in the form of `/controller/action`. In your case, I would just copy and modify the existing router to match your requirements. It's just 100 lines of code, probably less than many project's routing configuration XML or YAML file.

If the pages are content pages from a CMS or file templates, I usually add a ContentController with actions for each language and query the page: For example, `/content/en/about` would be routed to `ContentController->enAction('about')`.

**Comment by hispavista** [comment]
Ok. I needed it for one project, and I thougt that could be usefull. I will close pull request (Also this update is already in the master branch )

**Comment by lastzero** [comment]
Thank you for contributing!

**Pull request closed without merging**